app/support/auto_expire.rb

```diff
@@ -0,0 +1,44 @@
+class AutoExpire
+  def perform
+    order_details.each do |od|
+      od.transaction do
+        expire_reservation(od)
+      end
+    end
+  end
+
+  private
+
+  def order_details
+    purchased_active_order_details | non_reservation_order_details
+  end
+
+  def purchased_active_order_details
+    OrderDetail.purchased_active_reservations
+      .where("reservations.reserve_end_at < ?", Time.zone.now - 12.hours)
```

Comment:
I may need to stop by real quick today. I checked out the scopes on OrderDetail and saw a scope called `#in_progress_reservations` which uses `#comfirmed_reservation`.

I'm curious if we can re-use that or if that's different?